src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -526,16 +577,15 @@ private void ReadEndOfCentralDirectory()
                 // If the EOCD has the minimum possible size (no zip file comment), then exactly the previous 4 bytes will contain the signature
                 // But if the EOCD has max possible size, the signature should be found somewhere in the previous 64K + 4 bytes
                 if (!ZipHelper.SeekBackwardsToSignature(_archiveStream,
-                        ZipEndOfCentralDirectoryBlock.SignatureConstant,
-                        ZipEndOfCentralDirectoryBlock.ZipFileCommentMaxLength + ZipEndOfCentralDirectoryBlock.SignatureSize))
+                        ZipEndOfCentralDirectoryBlock.SignatureConstantBytes,
+                        ZipEndOfCentralDirectoryBlock.ZipFileCommentMaxLength + ZipEndOfCentralDirectoryBlock.SignatureConstantBytes.Length))
```

Comment:
Here and elsewhere: I believe you should now use `*.FieldLengths.Signature` instead of `*.SignatureConstantBytes.Length`.

Reply:
Yes, sorry. I didn't see these review comments - I'll roll them into the next commit with the remaining review!

Reply:
No worries. Ping me when you're done with all your updates and I'll run the mobile platforms leg.

Reply:
Thanks @carlossanlop. The main CI leg is running, and I'm ready for the mobile platforms leg.